js/pihole/settings.js

```diff
@@ -0,0 +1,70 @@
+$(function () {
+  $("[data-mask]").inputmask();
+});
+
+$(".confirm-reboot").confirm({
+	text: "Are you sure you want to send a reboot command to your Pi-Hole?",
+	title: "Confirmation required",
+	confirm(button) {
+		$("#rebootform").submit();
+	},
+	cancel(button) {
+		// nothing to do
+	},
+	confirmButton: "Yes, reboot",
+	cancelButton: "No, go back",
+	post: true,
+	confirmButtonClass: "btn-danger",
+	cancelButtonClass: "btn-success",
+	dialogClass: "modal-dialog modal-mg" // Bootstrap classes for mid-size modal
+});
+
+$(".confirm-restartdns").confirm({
+	text: "Are you sure you want to send a restart command to your DNS server?",
+	title: "Confirmation required",
+	confirm(button) {
+		$("#restartdnsform").submit();
+	},
+	cancel(button) {
+		// nothing to do
+	},
+	confirmButton: "Yes, restart DNS",
+	cancelButton: "No, go back",
+	post: true,
+	confirmButtonClass: "btn-danger",
+	cancelButtonClass: "btn-success",
+	dialogClass: "modal-dialog modal-mg"
+});
+
+$(".confirm-flushlogs").confirm({
+	text: "By default, the log is flushed at the end of the day via cron, but a very large log file can slow down the Web interface, so flushing it can be useful. Note that your statistics will be reset and you lose the statistics up to this point. Are you sure you want to flush your logs?",
+	title: "Confirmation required",
+	confirm(button) {
+		$("#flushlogsform").submit();
+	},
+	cancel(button) {
+		// nothing to do
+	},
+	confirmButton: "Yes, flush logs",
+	cancelButton: "No, go back",
+	post: true,
+	confirmButtonClass: "btn-danger",
+	cancelButtonClass: "btn-success",
+	dialogClass: "modal-dialog modal-mg"
+});
+
+$("#DHCPchk").click(function() {
+	$("input.DHCPgroup").prop("disabled", !this.checked);
+});
+
+$(document).ready(function() {
+	if(document.getElementById("DHCPLeasesTable"))
+	{
+		$("#DHCPLeasesTable").DataTable({
```

Comment:
I cannot solve this one. That is connected to how `jQuery DataTable` is defined. `codacy` just always finds something to blame you for.